src/monster.cpp

```diff
@@ -1348,6 +1381,44 @@ void monster::add_effect( const efftype_id &eff_id, int dur, body_part bp,
     Creature::add_effect( eff_id, dur, bp, permanent, intensity, force );
 }
 
+std::string monster::get_effect_status() const
+{
+    std::vector<std::string> effect;
+    if( has_effect( effect_downed ) ) {
+        effect.emplace_back( _( "On ground" ) );
+    }
+    if( has_effect( effect_stunned ) ) {
+        effect.emplace_back( _( "Stunned" ) );
+    }
+    if( has_effect( effect_lightsnare ) || has_effect( effect_heavysnare ) ||
+        has_effect( effect_beartrap ) ) {
+        effect.emplace_back( _( "Trapped" ) );
+    }
+    if( has_effect( effect_tied ) ) {
+        effect.emplace_back( _( "Tied" ) );
+    }
+    if( has_effect( effect_shrieking ) ) {
+        effect.emplace_back( _( "Shrieking" ) );
+    }
+    if( has_effect( effect_onfire ) ) {
+        effect.emplace_back( _( "On fire" ) );
+    }
+    if( has_effect( effect_in_pit ) ) {
+        effect.emplace_back( _( "In pit" ) );
+    }
+    if( has_effect( effect_grabbed ) ) {
+        effect.emplace_back( _( "Grabbed" ) );
+    }
+
```

Comment:
Instead of a giant `if` list, it would be better to add this as a property of effect type.
That way one could easily add a new visible effect without changing the code.